config/script_test.go

```diff
@@ -0,0 +1,308 @@
+					{
+						Commands: makeCommands(t, `
+						  [
+							["echo", tm_upper("hello ${global.some_string_var}")],
+							["stat", "."],
+						  ]
+						`),
+					},
+				},
+			},
+			globals: map[string]cty.Value{
+				"some_string_var": cty.StringVal("terramate"),
+			},
+			want: config.EvaluatedScript{
+				Labels:      labels,
+				Description: "some description",
+				Jobs: []config.EvaluatedScriptJob{
+					{
+						Commands: [][]string{
+							{"echo", "HELLO TERRAMATE"},
+							{"stat", "."},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "multiple jobs",
+			script: hcl.Script{
+				Labels: labels,
+				Description: hcl.NewScriptDescription(
+					makeAttribute(t, "description", `"some description"`)),
+				Jobs: []*hcl.ScriptJob{
+					{
+						Commands: makeCommands(t, `
+						  [
+							["echo", tm_upper("hello ${global.some_string_var}")],
+							["stat", "."],
+						  ]
+						`),
+					},
+					{
+						Commands: makeCommands(t, `
+						  [
+							["echo", tm_upper("hello ${global.some_string_var}")],
+							["ls", "-l"],
+						  ]
+						`),
+					},
+				},
+			},
+			globals: map[string]cty.Value{
+				"some_string_var": cty.StringVal("terramate"),
+			},
+			want: config.EvaluatedScript{
+				Labels:      labels,
+				Description: "some description",
+				Jobs: []config.EvaluatedScriptJob{
+					{
+						Commands: [][]string{
+							{"echo", "HELLO TERRAMATE"},
+							{"stat", "."},
+						},
+					},
+					{
+						Commands: [][]string{
+							{"echo", "HELLO TERRAMATE"},
+							{"ls", "-l"},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tcase := range tcases {
+		tcase := tcase
+		t.Run(tcase.name, func(t *testing.T) {
+			t.Parallel()
+			hclctx := eval.NewContext(stdlib.Functions(test.TempDir(t)))
+			hclctx.SetNamespace("global", tcase.globals)
+
+			got, err := config.EvalScript(hclctx, tcase.script)
+			assert.IsError(t, err, tcase.wantErr)
+
+			if tcase.want.Labels != nil {
+				assert.IsTrue(t, slices.Equal(tcase.want.Labels, got.Labels),
+					fmt.Sprintf("label mismatch: want[%#v], got [%#v]", tcase.want.Labels, got.Labels))
+			}
+			assert.EqualStrings(t, tcase.want.Description, got.Description)
+			assert.EqualInts(t, len(tcase.want.Jobs), len(got.Jobs), "len(jobs) mismatch")
+
+			for i, gotJob := range got.Jobs {
+				wantJob := tcase.want.Jobs[i]
+
+				if wantJob.Command != nil {
+					assert.IsTrue(t,
+						slices.Equal(wantJob.Command, gotJob.Command),
+						fmt.Sprintf("command value mismatch: want[%#v], got [%#v]", wantJob.Command, gotJob.Command))
+				}
+
+				if wantJob.Commands != nil {
+					assert.EqualInts(t, len(wantJob.Commands), len(gotJob.Commands), "len(commands) mismatch")
+					for j, gotCommand := range gotJob.Commands {
+						wantCommand := wantJob.Commands[j]
+						assert.IsTrue(t,
+							slices.Equal(wantCommand, gotCommand),
+							fmt.Sprintf("command value mismatch: want[%#v], got [%#v]", wantCommand, gotCommand))
+					}
+				}
+
+			}
```

Comment:
As `EvaluatedScript` and `EvaluatedScriptJob` are simple types (structs with basic Go types), what do you think of just using `cmp.Diff()` (go-cmp):

```
if diff := cmp.Diff(expectedScript, gotScript); diff != "" {
    t.Fatal(diff)
}
```

Reply:
Good idea, that would save us some lines of code and also the diff output is much more easier to read.